### PR TITLE
Use browserify for bundling JS modules

### PR DESCRIPTION
**Pull request by shrishrirang**
- Use browserify to bundle the modules instead of the home-grown custom bundling tool
- Remove submodules esprima and queryjs and use them as npm modules instead
- Remove custom code in InternalsVisible.js to expose modules by adding them to global namespace. Now, UTs directly reference the modules they need.
- Platform is defined as an alias (using remapify) that will map to the platform specific implementation depending on how it is defined at build time. Aliasing is needed as conditional 'require' has limitations with browserify.
- Remove unnecessary entry points from Gruntfile.js
- Remove nodeexports.js as it is not needed anymore
- Added some helper functions to Validate and Extensions modules
- $assert.areEqual implementation in the TestFrameworkAdapter now simply calls qunit's deepEqual instead of equal. This helps us perform comparisons on objects. WinJS test framework already does this.

### CODE REVIEW

**Comment by shrishrirang** [comment]
This addresses issue https://github.com/Azure/azure-mobile-services/issues/539

**Comment by shrishrirang** [comment]
Except few minor changes, this just ports https://github.com/Azure/azure-mobile-services/pull/815 from the offline branch.

File: sdk/package.json [hunk 1]
```diff
@@ -2,12 +2,20 @@
...
+    "grunt-browserify": "~4.0.1",
     "grunt-cli": "^0.1.13",
     "grunt-contrib-concat": "~0.5.0",
     "grunt-contrib-jshint": "~0.10.0",
     "grunt-contrib-nodeunit": "~0.4.1",
-    "grunt-contrib-uglify": "~0.6.0"
+    "grunt-contrib-uglify": "~0.6.0",
+    "grunt-contrib-watch": "~0.6.1",
+    "grunt-exec": "~0.4.6",
+    "query.js": "WindowsAzure/queryjs.git#02642e588694159a28db8ed736b4a4458c5bad9a",
```
Comment (phvannor) [inline sdk/package.json]: I think one of the issues outlines an issue using github .git urls here, not sure if another would work better?  Probably better than before for now.

File: sdk/src/Microsoft.WindowsAzure.Mobile.JS.csproj [hunk 2]
```diff
@@ -181,8 +180,8 @@
     <Content Include="MobileServices.intellisense.js" />
     <Content Include="MobileServices.priconfig.xml" />
     <Content Include="MobileServiceTable.js" />
-    <Content Include="Platforms\Platform.Web.js" />
-    <Content Include="Platforms\Platform.WinJS.js" />
+    <Content Include="Platforms\web\Platform.js" />
+    <Content Include="Platforms\winjs\Platform.js" />
```
Comment (phvannor) [inline sdk/src/Microsoft.WindowsAzure.Mobile.JS.csproj]: If we end up going with more .Web files (like Push was in the past) do they follow the same patterm> Push\web\push.js, etc?

File: sdk/src/Platforms/web/Platform.js [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
 /*global $__fileVersion__:false, $__version__:false */
 
-var _ = require('Extensions');
-var Validate = require('Validate');
-var Promises = require('Promises');
-var Resources = require('Resources');
+var _ = require('../../Utilities/Extensions');
+var Validate = require('../../Utilities/Validate');
+var Promises = require('../../Utilities/Promises');
+var Constants = require('../../Generated/Constants');
```
Comment (phvannor) [inline sdk/src/Platforms/web/Platform.js]: why rename resources to constants?  Seems kind of an odd name.

Comment (phvannor) [inline sdk/src/Platforms/web/Platform.js]: oh I see, you are moving resources down a level, still a lil naming confusion but cleaer to me now.

File: sdk/src/Microsoft.WindowsAzure.Mobile.JS.csproj (hunk 2 above)
Comment (shrishrirang) [inline sdk/src/Microsoft.WindowsAzure.Mobile.JS.csproj]: Yes, or may be have Platform\push.js.. either way. 
Do you foresee any trouble with this approach if the js-client is being consumed as a node package?

File: sdk/package.json (hunk 1 above)
Comment (shrishrirang) [inline sdk/package.json]: You mean using the full http:// link? Yeah, I can make that change

**Comment by danderson00** [comment]
Nothing stands out.

**Pull request merged**